evergage/evergage-global-templates
# Enhancement | Add stylelint configuration

# Description

Adds configuration for `stylelint`. All rules (including ordering) for the linter can be found in `.stylelintrc.json`

## Related Links
https://stylelint.io/user-guide/get-started
https://stylelint.io/user-guide/configure#rules
https://github.com/hudochenkov/stylelint-config-hudochenkov/blob/master/order.js

https://github.com/stylelint/stylelint
https://github.com/stylelint/stylelint-config-standard
https://github.com/hudochenkov/stylelint-order

## How to test
- run `npm i -g stylelint stylelint-config-standard stylelint-order`
- install the [`stylelint` VS Code extension](https://marketplace.visualstudio.com/items?itemName=stylelint.vscode-stylelint)
- add the following to `settings.json` in VS Code:
```
"editor.codeActionsOnSave": {
    "source.fixAll.styleLint": true
}
```
- view and save `.css` file for one of the templates
_(Note that since re-ordering CSS is not an auto-fixable problem in VS Code, it cannot be done on save. However, you should see that mis-ordered styles are underlined.)_

### TODOs
- update confluence doc ([this one?](https://confluence.internal.salesforce.com/pages/viewpage.action?pageId=245293245)) with instructions (or should this be added to the [dev docs](https://developer.evergage.com/templates/templates-style-guide-and-coding-conventions)?)
